### PR TITLE
Use generated cgo types for HTTP 

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Use types generated by  `go tool cgo -godefs` instead of relying upon `C.*` types.

### Motivation

This allows you to build system-probe, without having a full eBPF setup available to do the C compilation parts.

### Additional Notes

A post-processing script was added to convert the `char` arrays from `[]int8` to `[]byte` to allow easier interfacing. `int8` and `byte` are equivalently sized types in Go. This same post-processing is done in the `golang.org/x/sys` library.

```
$ go test -v -tags linux_bpf -run XXX -benchmem -bench BenchmarkPath
goos: linux
goarch: arm64
pkg: github.com/DataDog/datadog-agent/pkg/network/http
BenchmarkPath
BenchmarkPath-4   	83917098	        14.02 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/DataDog/datadog-agent/pkg/network/http	1.227s
$ go test -v -tags linux_bpf -run XXX -benchmem -bench BenchmarkOldPath
goos: linux
goarch: arm64
pkg: github.com/DataDog/datadog-agent/pkg/network/http
BenchmarkOldPath
BenchmarkOldPath-4   	30775982	        38.55 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/DataDog/datadog-agent/pkg/network/http	1.264s
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by p-lambert — APPROVED** [review]
left just 2 comments, but otherwise looks good to me :+1:

File: pkg/network/http/model.go [hunk 1]
```diff
@@ -9,100 +9,79 @@
...
+	k.Cpu = n.Cpu
+	k.Num = uint32(int(n.Idx) % HTTPBatchPages)
 }
 
 // Path returns the URL from the request fragment captured in eBPF with
 // GET variables excluded.
 // Example:
 // For a request fragment "GET /foo?var=bar HTTP/1.1", this method will return "/foo"
 func (tx *httpTX) Path(buffer []byte) ([]byte, bool) {
-	b := *(*[HTTPBufferSize]byte)(unsafe.Pointer(&tx.request_fragment))
```
[the remark this answers is not in the document]
Comment (brycekahle) [reply pkg/network/http/model.go:49]: Oh yes! I ran it, just forgot to include the results. Updated PR

[the remark this answers is not in the document]
Comment (brycekahle) [reply pkg/network/http/model.go:49]: It is actually faster 😁

File: pkg/network/http/http_types.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build ignore
+// +build ignore
+
+package http
```
[the remark this answers is not in the document]
Comment (brycekahle) [reply pkg/network/http/http_types.go:9]: This is the file we use to generate the Go types. I put it in this package, to reduce the amount of code changes necessary.

**Pull request merged**